clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -457,38 +461,32 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
             sensors.commitLatency.record(response.requestLatencyMs());
             for (Map.Entry<TopicPartition, Short> entry : commitResponse.responseData().entrySet()) {
                 TopicPartition tp = entry.getKey();
-                long offset = this.offsets.get(tp);
+                OffsetMetadata offsetMetadata = this.offsets.get(tp);
+                long offset = offsetMetadata.offset();
+
                 short errorCode = entry.getValue();
                 if (errorCode == Errors.NONE.code()) {
                     log.debug("Committed offset {} for partition {}", offset, tp);
                     if (subscriptions.isAssigned(tp))
                         // update the local cache only if the partition is still assigned
-                        subscriptions.committed(tp, offset);
-                } else if (errorCode == Errors.CONSUMER_COORDINATOR_NOT_AVAILABLE.code()
-                        || errorCode == Errors.NOT_COORDINATOR_FOR_CONSUMER.code()) {
-                    coordinatorDead();
-                    future.raise(Errors.forCode(errorCode));
-                    return;
-                } else if (errorCode == Errors.OFFSET_METADATA_TOO_LARGE.code()
```

Comment:
I just condensed the error checking here, since all we were doing is logging the errors. We'll still log them in the updated patch, but they didn't seem to require their own branch.